src/Mod/Arch/ArchIFC.py

```diff
@@ -64,19 +139,91 @@ def getIfcTypeSchema(self, IfcType):
         return None
 
     def getIfcSchema(self):
+        """Gets the IFC schema of all types relevant to this class.
+
+        Intended to be overwritten by the classes that inherit this class.
+
+        Returns
+        -------
+        dict
+            The schema of all the types relevant to this class.
+        """
+
         return {}
 
     def getCanonicalisedIfcTypes(self):
+        """This method gets the names of IFC types, converted to the form used in Arch.
+
+        This method changes the names of all IFC types to a more human readable
+        form which is used instead throughout Arch instead of the raw type
+        names. The names have the "Ifc" stripped from  the start of their
+        name, and spaces inserted between the words.
+
+        Returns
+        -------
+        list of str
+            The list of every IFC type name in their form used in Arch. List
+            will have names in the same order as they appear in the schema's
+            JSON, as per the .keys() method of dicts.
+
+        """
         schema = self.getIfcSchema()
         return [''.join(map(lambda x: x if x.islower() else " "+x, t[3:]))[1:] for t in schema.keys()]
 
     def getIfcAttributeSchema(self, ifcTypeSchema, name):
+        """Gets the schema of an IFC attribute with the given name.
+        
+        This method does so by converting the IFC attribute's name from the
+        human readable version Arch uses, and converting it to the less
+        readable name it has in the IFC schema.
+
+        Parameters
+        ----------
+        ifcTypeSchema: dict
+            The schema of the IFC type to access the attribute of.
+        name: str
+            The name the attribute has in Arch.
+
+        Returns
+        -------
+        dict:
```

Comment:
I think you don't need the final colon in this, because it's empty. It's just the type of the returned object, a `dict`.

Reply:
Resolved in 1ee0fde91b8d056ebadc7ac84ff1e639e7e6c906